jl4/src/L4/Parser.hs

```diff
@@ -362,26 +362,37 @@ declare sig =
 
 typeDecl :: Parser (TypeDecl Name)
 typeDecl =
-  recordDecl <|> enumDecl
+  recordDecl <|> enumOrSynonymDecl
```

Comment:
nit: I would prefer if this were
`recordDecl <|> enumDecl <|> synonymDecl`

Reply:
This is an artifact of LL(1) essentially. Both `enum` and `synonym` decls start with `IS`, and if you want to avoid unnecessary uses of `try`, you have to group them.